packages/wallet/src/data/services/alchemy/index.ts

```diff
@@ -1030,12 +1046,43 @@ async function _fetch<T extends ResponseBody>(
 
   if (!response.ok) {
     console.error(response.statusText)
+
+    if (response.status === 429) {
+      const apiKey = extractAlchemyApiKey(url)
+      if (apiKey) {
+        markApiKeyAsRateLimited(apiKey)
+      }
+    }
```

Comment:
non-blocking: Wouldn't be possible to extract the logic to some sort of `handleRateLimit` util and sharing it in the existing use cases, avoiding duplication?